src/main/java/com/github/manolo8/darkbot/extensions/plugins/PluginUpdater.java

```diff
@@ -0,0 +1,289 @@
+package com.github.manolo8.darkbot.extensions.plugins;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.gui.plugins.UpdateProgressBar;
+import com.github.manolo8.darkbot.gui.utils.Popups;
+import com.github.manolo8.darkbot.utils.Time;
+
+import javax.swing.*;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+//todoo i18n
+public class PluginUpdater {
+
+    private Date lastChecked = new Date();
+
+    public final List<Plugin> UPDATES = new ArrayList<>();
```

Comment:
Just avoid holding the list to plugins, use pluginHandler.getPlugins() and ignore the plugins update statuses that aren't AVAILABLE